ratpack-core/src/main/java/ratpack/http/client/Proxy.java

```diff
@@ -39,6 +41,22 @@ public interface Proxy {
    */
   int getPort();
 
+  /**
+   * The username to use when connecting to the proxy.
+   *
+   * @return The username to use when connecting to the HTTP proxy, or null if a username should not be used when connecting to the HTTP proxy.
+   */
+  @Nullable
+  String getUsername();
```

Comment:
I considered using `Optional<String>`, but it looks like nullable fields and null-checks are more common for user-provided values in Ratpack's public API.

Reply:
ACK.